xarray/core/common.py

```diff
@@ -938,6 +938,40 @@ def ones_like(other, dtype=None):
     return full_like(other, 1, dtype)
 
 
+def structured_like(data, other):
+    """Return a new object labeled in the same way as the given object.
+
+    Parameters
+    ----------
+    data : array_like
+        Data to fill the new object. Must have same shape as other.
+    other : DataArray, or Variable
+        The reference object from which to get labels
+
+    Returns
+    -------
+    out : same as other
+        New object with the same shape and properties as other.
+        Coords will be copied from other.
```

Comment:
Maybe: "New object with dimensions, attributes, coordinates and name copied from template."